server/app/services/rpc_server.rb

```diff
@@ -25,8 +28,38 @@ def initialize(code, message, backtrace = nil)
 
   attr_reader :handlers
 
-  def initialize
+  def initialize(queue)
+    @queue = queue
     @handlers = {}
+    @counter = 0
+    @processing = false
+    every(REPORT_EVERY) {
+      report_queue
+    }
+  end
+
+  def report_queue
+    if @queue.size > QUEUE_WARN_LIMIT
+      warn "#{@queue.size} messages in queue"
+      info "#{@counter / REPORT_EVERY} requests per second"
+    end
+    @counter = 0
+  end
+
+  def process!
+    @processing = true
+    defer {
```

Comment:
Agreed, I'll try to refactor this reporting to `WebsocketBackend` (although numbers there are different).